models/tags.go

```diff
@@ -45,12 +45,31 @@ type CreateTagsSchema struct {
 	EntityName string      `json:"entity_name"`
 }
 
+type CreateNewTagSchema struct {
+	Name       string `json:"name" binding:"required,min=1,max=20"`
+	Color      string `json:"color"`
+	EntityType string `json:"entity_type"`
+	EntityName string `json:"entity_name"`
+}
+
 type RemoveTagsSchema struct {
 	Names      []string `json:"names"`
 	EntityType string   `json:"entity_type"`
 	EntityName string   `json:"entity_name"`
 }
 
+type EditTagsSchema struct {
+	TagsToAdd    []Tag              `json:"tags_to_add"`
+	TagsToRemove []string           `json:"tags_to_remove"`
+	EntityType   string             `json:"entity_type"`
+	EntityID     primitive.ObjectID `json:"entity_id"`
+}
+
 type GetTagsSchema struct {
 	EntityType string `json:"entity_type"`
 }
+
+type GetTagsByEntityNameSchema struct {
+	EntityType string `json:"entity_type"`
```

Comment:
if this is by name so why type is also a parameter

Reply:
schema, station and user can all have the same name

Reply:
but this endpoint not in use - I deleted it